inc/packages/namespace.php

```diff
@@ -837,4 +874,96 @@ function fetch_and_validate_package_alias( DIDDocument $did ) {
 	return $domain;
 }
 
+/**
+ * Enable searching by DID.
+ *
+ * @param mixed  $result The result of the plugins_api call.
+ * @param string $action The action being performed.
+ * @param stdClass $args The arguments passed to the plugins_api call.
+ * @return mixed The search result for the DID.
+ */
+function search_by_did( $result, $action, $args ) {
+	if ( 'query_plugins' !== $action || empty( $args->search ) ) {
+		return $result;
+	}
+
+	// The DID comes from a URL-encoded request parameter, and must be decoded first.
+	$did = sanitize_text_field( urldecode( $args->search ) );
+	if ( ! str_starts_with( $did, 'did:plc:' ) || strlen( $did ) !== 32 ) {
+		return $result;
+	}
+
+	$api_data = get_api_data( $did );
+	if ( is_wp_error( $api_data ) ) {
+		return $result;
+	}
+
+	$result = [
+		'plugins' => [ $api_data ],
+		'info' => [
+			'page' => 1,
+			'pages' => 1,
+			'results' => 1,
+			'total' => 1,
+		],
+	];
+
+	return (object) $result;
+}
+
+/**
+ * Get API data for a DID.
+ *
+ * @param string $did DID.
+ * @return array|WP_Error The API data array or WP_Error on failure.
+ */
+function get_api_data( $did ) {
+	$api_data = get_update_data( $did );
+	if ( is_wp_error( $api_data ) ) {
+		return $api_data;
+	}
+
+	$api_data = json_decode( json_encode( $api_data ), true );
+	$api_data['description'] = $api_data['sections']['description'];
+	$api_data['short_description'] = substr( strip_tags( trim( $api_data['description'] ) ), 0, 147 ) . '...';
+	$api_data['last_updated'] ??= 0;
+	$api_data['num_ratings'] ??= 0;
+	$api_data['rating'] ??= 0;
+	$api_data['active_installs'] ??= 0;
+
+	// Avoid a double-hashed slug.
+	$hash_suffix = '-' . get_did_hash( $did );
+	if ( str_ends_with( $api_data['slug'], $hash_suffix ) ) {
+		$api_data['slug'] = str_replace( $hash_suffix, '', $api_data['slug'] );
+	}
+
+	return $api_data;
```

Comment:
Should we rename get_update_data to something like get_package_data() and just add 

- description
- short_description
- last_updated
- num_ratings
- rating
- active_installs

We could then simplify this function.

Reply:
Or we could add a `slug_didhash` to the data returned from the new `get_package_data()` then just use that as needed.

Reply:
Yeah we could add the slug_didhash too so it's available without needing to call get_did_hash() in each location, though I'd also be happy enough if we just had slug and hashed it whenever needed. No strong feelings one way or the other.